File: java/src/main/java/com/workshop/architecture/fitness/clean_architecture/use_case/port/outbound/ForStoringBillingReferences.java
package com.workshop.architecture.fitness.clean_architecture.use_case.port.outbound;

import com.workshop.architecture.fitness.clean_architecture.entity.MembershipBillingReference;

public interface ForStoringBillingReferences {
    MembershipBillingReference storeMembershipBillingReference(MembershipBillingReference billingReference);
}

File: python/src/workshop_api/fitness/clean_architecture/use_case/port/outbound/for_storing_billing_references.py
from typing import Protocol

from workshop_api.fitness.clean_architecture.entity.membership_billing_reference import (
    MembershipBillingReference,
)


class ForStoringBillingReferences(Protocol):
    def store_membership_billing_reference(
        self,
        billing_reference: MembershipBillingReference,
    ) -> MembershipBillingReference:
        pass
